coral-erm/coral
# Upgrade file uploads

This PR fixes issues #475 and #458.

    This patch adds meaningful error messages when an upload fails for any reason.
    
    This is more a rewrite than a bugfix: it replaces the use of AjaxUpload
    (code from 2008, apparently not maintained anymore) with a standard JQuery
    ajax upload.
    
    It handles both PHP upload errors
    (see https://www.php.net/manual/en/features.file-upload.errors.php)
    and webserver errors ("413 Entity Too Large" for example)
    
    In order to have standard JQuery ajax upload, jquery needs to be upgraded.
    
    This patch upgrades jquery from 1.4.4 to 1.8.0
    
    The original datepicker was not compatible with jquery 1.8.0, so it has been
    updated with a fork of the existing one:
    https://github.com/michaelulguim/jQuery-Datepicker--Kevin-Luck-
    It's not a new datepicker, it's an update of the existing datepicker.
    
    Some cleanup has also been done:
     - ajaxupload plugin has been removed
     - unused older jquery versions were remove

The following file uploads have been updated:
 - Attachments to resources
 - Attachments and documents to licenses
 - Attachments and documents in management.

Test plan:
 - Apply the patch
 - Try uploading a file that exceeds the maximum size allowed by your webserver or php configuration
 - Check that you got an error message
 - Check that datepickers still work.

![Capture d’écran_2019-12-11_16-49-26](https://user-images.githubusercontent.com/1679997/70636660-374fa680-1c36-11ea-8643-80852e4b05e9.png)
